onmt/opts.py

```diff
@@ -643,30 +644,42 @@ def _add_decoding_opts(parser):
                    "target token. If it is not provided (or the identified "
                    "source token does not exist in the table), then it "
                    "will copy the source token.")
+    group.add('--ban_unk_token', '-ban_unk_token',
+              action="store_true",
+              help="Prevent unk token use by setting unk proba to 0")
     group.add('--phrase_table', '-phrase_table', type=str, default="",
               help="If phrase_table is provided (with replace_unk), it will "
                    "look up the identified source token and give the "
                    "corresponding target token. If it is not provided "
                    "(or the identified source token does not exist in "
                    "the table), then it will copy the source token.")
+    group.add('--beam_size', '-beam_size', type=int, default=5,
+              help='Beam size')
 
     group = parser.add_argument_group('Random Sampling')
     group.add('--random_sampling_topk', '-random_sampling_topk',
-              default=1, type=int,
+              default=0, type=int,
               help="Set this to -1 to do random sampling from full "
                    "distribution. Set this to value k>1 to do random "
                    "sampling restricted to the k most likely next tokens. "
                    "Set this to 1 to use argmax or for doing beam "
                    "search.")
+    group.add('--random_sampling_topp', '-random_sampling_topp',
+              default=0, type=float,
+              help="Probability for top-p/nucleus sampling. Restrict tokens"
+                   " to the most likely until the cumulated probability is"
+                   " over p. In range [0, 1]."
+                   " https://arxiv.org/abs/1904.09751")
     group.add('--random_sampling_temp', '-random_sampling_temp',
               default=1., type=float,
               help="If doing random sampling, divide the logits by "
                    "this before computing softmax during decoding.")
+    group.add('--always_sample_eos', '-always_sample_eos',
```

Comment:
What are the cases where we don't want to sample eos? Shouldn't this be handled via `min_length` instead?

Reply:
This is specific to topk/topp. Even if eos token is not in the topk/topp tokens, we keep it rather than excluding it.
If there is a min_length, eos will still be removed, always_sample_eos does not overrule min_length.

Reply:
I will remove it, I don't think we will really need this option